### PR TITLE
Add homepage to github:publish options

### PR DESCRIPTION
**Pull request by leon-vg**
Signed-off-by: Leon van Ginneken <leon.van.ginneken@alliander.com>

## Hey, I just made a Pull Request!

I would like to add a 'homepage' when publishing a new Github repository through a Software Template. I think this should do the trick.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/scaffolder-backend/api-report.md [hunk 1]
```diff
@@ -187,7 +188,7 @@ export function createGithubRepoCreateAction(options: {
   allowMergeCommit?: boolean | undefined;
   requireCodeOwnerReviews?: boolean | undefined;
   requiredStatusCheckContexts?: string[] | undefined;
-  repoVisibility?: 'internal' | 'private' | 'public' | undefined;
+  repoVisibility?: 'public' | 'private' | 'internal' | undefined;
```
Comment (fearphage) [inline plugins/scaffolder-backend/api-report.md:191]: All of these similar changes were previously in alphabetical order. Why the change?

Comment (leon-vg) [reply plugins/scaffolder-backend/api-report.md:191]: I noticed this too. This is the result of running `yarn build:api-reports:only plugins/scaffolder-backend`. But I guess something is still off, because the tests on this file are still failing. Not sure what it is yet.

I ran `yarn build:api-reports:only plugins/scaffolder-backend` because running `yarn build:api-reports:only` results in this error:

```
# Generating package API reports
## Processing plugins\user-settings
Error: The API Report for plugins\user-settings is not allowed to have warnings
```

Comment (Rugvip) [reply plugins/scaffolder-backend/api-report.md:191]: Hmm, will have to look into this separately, but for now you'll need to run the full `yarn build:api-reports`

File: .changeset/beige-pumas-tap.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-scaffolder-backend': minor
+---
+
+Add the option for a homepage when using the github:publish action
```
Comment (Rugvip) [inline .changeset/beige-pumas-tap.md:5]:
```suggestion
Add the option for a homepage when using the `github:publish` action
```

**Review by Rugvip — APPROVED** [review]
Thank you, looks good! 👍 

Just need the API report fixed, as mentioned `yarn build:api-reports` should do the trick. There's likely some repo-global type state that messes up the report that we'll need to narrow down on our side.

File: plugins/scaffolder-backend/api-report.md (hunk 1 above)
Comment (leon-vg) [reply plugins/scaffolder-backend/api-report.md:191]: I keep struggling with the `build:api-reports` command. It keeps giving me errors on other plugins that are not allowed to have warnings. Any clues?

```
$ tsc --project C:\DEV\backstage-original\tsconfig.json --skipLibCheck false --incremental false
# Generating package API reports
## Processing plugins\xcmetrics
Error: The API Report for plugins\xcmetrics is not allowed to have warnings
    at runApiExtraction (C:\DEV\backstage-original\scripts\api-extractor.ts:562:13)
    at async main (C:\DEV\backstage-original\scripts\api-extractor.ts:1360:5)
```

Comment (leon-vg) [reply plugins/scaffolder-backend/api-report.md]: It takes a really long time to run. Perhaps this is due to a timeout?

Anyway. I ran `yarn build:api-reports plugins/scaffolder-backend` again, because that does give results. It reverted some of the earlier changes, but still fails the test unfortunately. I think because the order of one property is still changed in this PR.

Comment (Rugvip) [reply plugins/scaffolder-backend/api-report.md]: Alright, dunno what might by up tbh, gonna verify that master isn't broken. I'll ship a fix to your PR in a bit too

Comment (Rugvip) [reply plugins/scaffolder-backend/api-report.md]: Tried building API reports on latest master and it seems all good. Most likely something with your local environment that's up. You've ran `yarn install` as well right?

Either way I pushed up an updated API report so hopefully should be good to go now

Comment (leon-vg) [reply plugins/scaffolder-backend/api-report.md]: Thanks! I did indeed `yarn install` and then `yarn build:api-reports`. Not sure what it is. But happy that you fixed it! 👍
